src/compiler/checker.ts

```diff
@@ -14406,12 +14437,113 @@ namespace ts {
             }
         }
 
+        function checkUnusedIdentifiers(node: FunctionDeclaration | MethodDeclaration | ConstructorDeclaration | FunctionExpression | ArrowFunction | ForInStatement | Block | CatchClause): void {
+            if (node.parent.kind !== SyntaxKind.InterfaceDeclaration && (compilerOptions.noUnusedLocals || compilerOptions.noUnusedParameters) && !isInAmbientContext(node)) {
+                for (const key in node.locals) {
+                    if (hasProperty(node.locals, key)) {
+                        const local = node.locals[key];
+                        if (!local.hasReference && local.valueDeclaration) {
+                            if (local.valueDeclaration.kind !== SyntaxKind.Parameter && compilerOptions.noUnusedLocals) {
+                                error(local.valueDeclaration, Diagnostics._0_is_declared_but_never_used, key);
+                            }
+                            else if (local.valueDeclaration.kind === SyntaxKind.Parameter && compilerOptions.noUnusedParameters) {
+                                if (local.valueDeclaration.modifiers) {
+                                    if (getCombinedNodeFlags(local.valueDeclaration) & NodeFlags.Private) {
+                                        error(local.valueDeclaration, Diagnostics._0_is_declared_but_never_used, key);
+                                    }
+                                }
+                                else {
+                                    error(local.valueDeclaration, Diagnostics._0_is_declared_but_never_used, key);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        function checkUnusedModuleLocals(node: ModuleDeclaration): void {
+            if (compilerOptions.noUnusedLocals && !isInAmbientContext(node)) {
+                for (const key in node.locals) {
+                    if (hasProperty(node.locals, key)) {
+                        const local = node.locals[key];
+                        if (!local.hasReference && !local.exportSymbol) {
+                            if ((local.valueDeclaration && local.valueDeclaration.kind)) {
+                                error(local.valueDeclaration, Diagnostics._0_is_declared_but_never_used, key);
+                            }
+                            else if (local.declarations &&

```

Comment:
I have already tried this one. But I feel this might not work. The reason is that ImportEqualsDeclaration is a special case in which we need to use local.declarations[0].name instead of local.declarations[0]. 

Scenario:
module A {
    export class Calculator {
        public handelChar() {
        }
    }
}

module B {
    import a = A;
}

Here 'a' must be indicated as error instead of "import a = A".  But, I have optimized it a little so the condition does not look long.